internal/auth/auth_client.go

```diff
@@ -0,0 +1,69 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	authclient "github.com/fabric8-services/fabric8-jenkins-idler/internal/auth/client"
```

Comment:
I understand that `/client` is an autogenerated package by Goa, so you won't find it on GitHub and thus you can't do import from github url, but we need a better way than this. Otherwise everytime auth service changes it's code we will have to as well.

Plus this is adding way too much unnecessary code. Why do you need this client? Is this for getting service account token?

Reply:
> I understand that /client is an autogenerated package by Goa, so you won't find it on GitHub and thus you can't do import from github url, but we need a better way than this. Otherwise everytime auth service changes it's code we will have to as well.

You are loosing me here. What is your issue? And what is the alternative really? This is
how code generation works. This code is generated as part of the build and then checked in. This is not different than checking in the lock file for Dep. This ensures that the code still compiles when you try to check it out at a later stage. And yes, if Auth would be changing, you would need to regenerate the client (provided it changes in a incompatible way which generally should be avoided by any service).

So we run the generation now to get the current client. This code is checked in, so generation is not going to happen on each build. If time comes where either Auth changes or we want to for some other reasons update to the latest Auth version, we would delete the client directory and re-rebuild at which stage the latest client gets created and one might have to make code adjustments. Really, not so different to dependnecy management at all.

> Plus this is adding way too much unnecessary code. Why do you need this client? 

Because we need to talk to Auth. So yes, we can continue with hand wiring our client, but this does not really scale. I thought that we already agreed that Goa is the way forward. You are getting more types interfaces and you also get a generic handling of the various errors which can occur using HTTP as transport.

The code is by no means unnecessary. I would not have added it, if it would not be used. We might not use all of the Auth API, but we use it to get the OISIO token and the cluster view. These pieces are essential for multi-cluster support.

Reply:
I get this now. I was under an impression that this is separately generated from fabric8-auth service and copied here. Just went through makefile. Sorry about that.
But this package(`internal/auth/client`) should be added to .gitingnore
I don't see any harm in building it every time.

Reply:
No, if you add it to .gitignore you cannot grantee reproducible builds. See comment above. Say you are not checking it in and write your code against the generated code. Now you commit and move on. At some stage you want to go back and revisit something and you want to build the code at this specific commit. If in the meantime the Auth Service changed (the generation is against master), you might get a broken build. This is exactly the same as the lock file for Dep.

Reply:
> I don't see any harm in building it every time.

See above.

Reply:
Agreed with @hferentschik this is standard practice in go world and/or using go's dep or glide